File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
       int p1=0;
        int p2=0;
        while(p1<s.size()){
            while(p2 < s.size() && s[p2] !=' ')
                p2++;
            reverse(s.begin()+p1,s.begin()+p2);
               
            p1=p2+1;
            p2=p1;
        }
        return s;
    }
};

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        char[] c=s.toCharArray();
        int i=0;
        for(int j=0;j<c.length;j++)
        {
            if(c[j]==' ')
            {
                reverse(c,i,j-1);
                i=j+1;
              
            }
        }
        reverse(c,i,c.length-1);
        return new String(c);
    }
    public void reverse(char[] c,int i,int j)
    {
        while(i<=j)
        {
            char t=c[i];
            c[i]=c[j];
            c[j]=t;
            i++;
            j--;
        } 
    }
}